web/packages/teleport/src/AuthConnectors/AuthConnectorTile.tsx

```diff
@@ -0,0 +1,227 @@
+import React from 'react';
+import { useTheme } from 'styled-components';
+
+import { Flex, H2, Subtitle2, ButtonSecondary, P3, Label } from 'design';
+import {
+  ArrowRight,
+  CircleCheck,
+  MinusCircle,
+  Password,
+  RocketLaunch,
+} from 'design/Icon';
+
+import { MenuIcon, MenuItem } from 'shared/components/MenuAction';
+import { AuthType } from 'shared/services';
+
+import { State as ResourceState } from 'teleport/components/useResources';
+import { ConnectorBox } from 'teleport/AuthConnectors/styles/ConnectorBox.styles';
+import { ButtonLockedFeature } from 'teleport/components/ButtonLockedFeature';
+import { CtaEvent } from 'teleport/services/userEvent';
+
+export function AuthConnectorTile({
+  id,
+  name,
+  kind,
+  Icon,
+  isDefault,
+  isPlaceholder,
+  onSetup,
+  isEnabled,
+  customDesc,
+  onEdit,
+  onDelete,
+  isCTA = false,
+}: Props) {
+  const theme = useTheme();
+  const onClickEdit = () => onEdit(id);
+  const onClickDelete = () => onDelete(id);
+
+  let desc;
+  switch (kind) {
+    case 'github':
+      desc = 'GitHub Connector';
+      break;
+    case 'oidc':
+      desc = 'OIDC Connector';
+      break;
+    case 'saml':
+      desc = 'SAML Connector';
+      break;
+  }
+
+  const ActionsSection = () => {
+    if (!isCTA && !isPlaceholder) {
+      return <EnabledIndicator isEnabled={isEnabled} />;
+    }
+
+    // If this is a placeholder tile, only show a "Set Up" button.
+    if (isPlaceholder && !!onSetup) {
+      return (
+        <ButtonSecondary
+          onClick={onSetup}
+          px={3}
+          css={`
+            position: absolute;
+          `}
+        >
+          Set Up <ArrowRight size="small" ml={2} />
+        </ButtonSecondary>
+      );
+    }
+
+    if (isCTA) {
+      return (
+        <ButtonLockedFeature
+          event={CtaEvent.CTA_AUTH_CONNECTOR}
+          noIcon
+          css={`
+            border-radius: 62px;
+            font-weight: 400;
+            width: 185px;
+            height: 32px;
+            position: absolute;
+          `}
+          size="small"
+          px={3}
+          py={2}
+        >
+          <RocketLaunch size="small" mr={2} />
+          Upgrade to Enterprise
+        </ButtonLockedFeature>
+      );
+    }
+  };
+
+  return (
+    <ConnectorBox>
+      <Flex
+        flexDirection="column"
+        justifyContent="space-between"
+        alignItems="flex-start"
+        height="100%"
+      >
+        <Icon />
+        <Flex flexDirection="column" alignItems="flex-start" gap={1}>
+          <Flex alignItems="center" gap={2}>
+            <H2>{name}</H2>
+            {isDefault && (
+              <Label
+                kind="secondary"
+                css={`
+                  color: ${props => props.theme.colors.text.slightlyMuted};
+                `}
+              >
+                Default
+              </Label>
+            )}
+          </Flex>
+          <Subtitle2 color="text.slightlyMuted">{customDesc || desc}</Subtitle2>
```

Comment:
Are users able to provide `customDesc`? If so, it seems that with the `height` set to a specific value, this will make the content overflow. I assume we'd like to show ellipsis instead.

To do so, you can set this on this `p`:

```css
display: -webkit-box;
-webkit-box-orient: vertical;
-webkit-line-clamp: 1;
```

We use this [in notifications](https://github.com/gravitational/teleport/blob/d87a04d28e83ac73b6350d340582f3115be29a50/web/packages/shared/components/Notification/Notification.tsx#L288-L293) already. This works even when `height` is not set to a specific value.

If users are able to provide their own desc, it'd be a good idea to add a tile in the story that has long values to test if everything is rendered properly.

Reply:
This isn't a user-configurable field, it's just a way for the parent of this component to hardcode a description, for example for a SAML CTA it's "Okta, OneLogin, Azure ..."